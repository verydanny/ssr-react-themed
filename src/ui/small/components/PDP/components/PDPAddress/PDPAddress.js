import React from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed-too'

const PDPAddress = ({
  city,
  state,
  zipCode,
  neighborhood,
  streetAddress,
  theme,
}) => (
  <div
    className={theme.PDPAddress}
    data-tid="pdpAddress"
    itemScope
    itemType="https://schema.org/PostalAddress"
    itemProp="address"
  >
    <div
      className={theme.PDPAddress_StreetAddress}
      data-tid="addressStreetAddress"
      itemProp="streetAddress"
    >
      {streetAddress}
    </div>
    <div data-tid="addressCityStateZip">
      <span itemProp="addressLocality">{city}</span>{city && ', '}
      <span itemProp="addressRegion">{state}</span>{' '}
      <span itemProp="postalCode">{zipCode}</span>
    </div>
    <div data-tid="addressNeighborhood">{neighborhood}</div>
  </div>
)

PDPAddress.propTypes = {
  city: PropTypes.string,
  state: PropTypes.string,
  zipCode: PropTypes.string,
  neighborhood: PropTypes.string,
  streetAddress: PropTypes.string,
  theme: PropTypes.object,
}

PDPAddress.defaultProps = {
  theme: {},
}

export default themed(/^PDPAddress/, { pure: true })(PDPAddress)
