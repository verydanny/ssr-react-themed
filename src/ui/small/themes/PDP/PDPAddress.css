@value ( RentBlue, Gray1 ) from 'ui/shared/themes/colors/colors.css';

.PDPAddress {
  color: Gray1;
  font-size: 1em;
  line-height: 1.75em;
  padding: 0.5em;
  text-align: center;
  word-break: break-word;
}

.PDPAddress_StreetAddress {
  color: RentBlue;
  font-size: 1.25em;
}
