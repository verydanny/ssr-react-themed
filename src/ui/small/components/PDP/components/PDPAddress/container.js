import { connect } from 'react-redux'
import {
  getAddress,
  getCity,
  getState,
  getZipCode,
  getNeighborhood,
} from 'app/selectors/pdpListing'
import PDPAddress from './PDPAddress'

const mapStateToProps = state => ({
  city: getCity(state),
  state: getState(state),
  zipCode: getZipCode(state),
  neighborhood: getNeighborhood(state),
  streetAddress: getAddress(state),
})

export default connect(mapStateToProps)(PDPAddress)
