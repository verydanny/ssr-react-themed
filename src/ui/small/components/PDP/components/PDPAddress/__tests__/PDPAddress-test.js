import React from 'react'
import { mount } from 'enzyme'
import { keyMirror } from '@rentpath/react-ui-utils'
import PDPAddress from '../PDPAddress'

const theme = keyMirror([
  'PDPAddress',
  'PDPAddress_StreetAddress',
])

const defaultProps = {
  theme,
}

const factory = props =>
  <PDPAddress {...defaultProps} {...props} />

describe('<PDPAddress />', () => {
  it('should render streetAddress', () => {
    const city = 'Atlanta'
    const state = 'GA'
    const zipCode = '30318'
    const cityStateZip = `${city}, ${state} ${zipCode}`
    const neighborhood = 'Reynoldstown'
    const streetAddress = '841 Memorial Dr SE'

    const wrapper = mount(factory({
      city,
      state,
      zipCode,
      neighborhood,
      streetAddress,
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.PDPAddress_StreetAddress').text()).toEqual(streetAddress)
    expect(wrapper.find('[data-tid="addressCityStateZip"]').text()).toEqual(cityStateZip)
    expect(wrapper.find('[data-tid="addressNeighborhood"]').text()).toEqual(neighborhood)
  })

  it('should have a theme', () => {
    const wrapper = mount(factory({ theme }))

    expect(wrapper.props().theme).toBeTruthy()
  })
})
