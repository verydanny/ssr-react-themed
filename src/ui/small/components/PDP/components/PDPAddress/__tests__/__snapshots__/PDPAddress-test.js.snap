// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PDPAddress /> should render streetAddress 1`] = `

<div class="PDPAddress"
     data-tid="pdpAddress"
     itemscope
     itemtype="https://schema.org/PostalAddress"
     itemprop="address"
>
  <div class="PDPAddress_StreetAddress"
       data-tid="addressStreetAddress"
       itemprop="streetAddress"
  >
    841 Memorial Dr SE
  </div>
  <div data-tid="addressCityStateZip">
    <span itemprop="addressLocality">
      Atlanta
    </span>
    ,
    <span itemprop="addressRegion">
      GA
    </span>
    <span itemprop="postalCode">
      30318
    </span>
  </div>
  <div data-tid="addressNeighborhood">
    Reynoldstown
  </div>
</div>

`;
